app/Providers/ValidationServiceProvider.php

```diff
@@ -96,6 +100,24 @@ public function boot()
         });
 
 
+        /**
+         * This is the validator replace static method that allows us to pass the $parameters of the table names
+         * into the translation string in validation.two_column_unique_undeleted for two_column_unique_undeleted
+         * validation messages.
+         *
+         * This is invoked automatically by Validator::extend('two_column_unique_undeleted') above and
+         * produces a translation like: "The name value must be unique across categories and category type."
+         */
+        Validator::replacer('two_column_unique_undeleted', function($message, $attribute, $rule, $parameters) {
+            $message = str_replace(':table1', $parameters[0], $message);
+            $message = str_replace(':table2', $parameters[2], $message);
```

Comment:
Wouldn't mind a wee comment here explaining why the jump from 0 to 2?

Reply:
I didn't bother, since the existing Validator uses those parameters, but I can add it in.